### DISCUSSION TITLE
How to config `Telescope` support searching for selected texts in visual mode?

### DISCUSSION THREAD

**Discussion by JevanWu** [NvChad/NvChad#2321] [Q&A] [answered]
Hi, I have a request to set a shortcut to search for selected texts. How to configure this in NvChad? 
I've tried some solutions on the Internet. However, they didn't work pretty well.

Now. I'm using 
```
  v = {
    ["<C-f>"] = { "y:Telescope live_grep default_text=<C-r>0 <CR>", "Find selected text" }
  }
```
But it seems to only support one word rather than all of the selected texts

Someone can help? Thanks in advance~

**Comment by siduck** [comment]
what is y:? it should b either <cmd> or just :

**Reply by JevanWu** [reply]
I found the solution [here](https://www.reddit.com/r/neovim/comments/p8wtmn/telescopenvim_how_to_take_what_i_selected_in/). The `y` here is the `yank`

**Reply by JevanWu** [reply]
But nvim would throw an error if I selected a sentence when I was using this solution

**Reply by georgejean** [reply, marked as answer]
I will not solve your problem for multiline selection but I'll just give one remark.
You can replace your mapping by :
```lua
v = {
    ["<C-f>"] = { "<cmd> Telescope grep_string <CR>", "Find selected text" },
  },
```
because it uses the selected text for the search (in normal mode, it's the word under cursor). It doesn't seem to work for multiline search but it gives no error and it doesn't affect the yank register.

**Reply by JevanWu** [reply]
@georgejean really Thank you. It looks like a good solution. But I wonder why it works sometimes and it doesn't work sometimes. i.e. 
<img width="868" alt="image" src="https://github.com/NvChad/NvChad/assets/4226011/51d8841a-b3c3-4116-b34b-c5eff50e0465">
For this case, `telescope` failed to search for the text

**Reply by georgejean** [reply]
@JevanWu
I can't reproduce it but I see that you can't look for files inside custom folder which is gitignored (it's the default for ripgrep).
I guess you should perhaps update NvChad so that you get the file `.ignore` in lua folder which enables searching for files inside custom folder. Do you have already this file?

**Reply by JevanWu** [reply]
Ah. You're right. Thank you very much~
